### DISCUSSION TITLE
Losing precision on decimal number during conversion from JSON

### DISCUSSION THREAD

**Discussion by trunklegs911** [Cinchoo/ChoETL#287] [Q&A] [answered]
I have a JSON string with a decimal value 26.61657491, and when I write the JSON file to parquet:

`using(var r = ChoJSONReader.LoadText(jsonString)) {
    using var writer = new ChoParquetWriter(filepath);
    writer.Write(r);
}`

which appears to be successful, but when I open the resulting parquet file the decimal is now 26.616575. So it has been rounded down somewhere in the conversion process. 

Is this expected behavior of writing parquet files? Is there usually going to be some loss of precision in this way? 

Thanks!

**Comment by Cinchoo** [comment, marked as answer]
It seems to be bug in the library. While I fix and release it, you can do as below in the meantime to fix this issue

```
            using (var r = ChoJSONReader.LoadText(jsonString)
                .UseJsonSerialization()
                )
            { 
                using (var writer = new ChoParquetWriter(filepath))
                    writer.Write(r); 
            }
```

**Reply by trunklegs911** [reply]
Thank you!
